controllers/openstacknetconfig_controller.go

```diff
@@ -819,14 +815,41 @@ func (r *OpenStackNetConfigReconciler) createOrUpdateOpenStackMACAddress(
 	}
 
 	//
-	// get all VMset
+	// get all IPsets
 	//
-	labelSelector := map[string]string{}
+	labelSelector := map[string]string{
+		openstacknetconfig.OpenStackNetConfigReconcileLabel: instance.Name,
+	}
 	listOpts := []client.ListOption{
 		client.InNamespace(instance.Namespace),
 		client.MatchingLabels(labelSelector),
 	}
 
+	ipSetList := &ospdirectorv1beta1.OpenStackIPSetList{}
+	if err := r.GetClient().List(
+		context.Background(),
+		ipSetList,
+		listOpts...,
+	); err != nil && !k8s_errors.IsNotFound(err) {
+		return macAddress, err
```

Comment:
Would it be safer to return `nil` here instead of `macAddress`.  I guess it doesn't really matter, since the caller should see the non-nil `err` and act on that.

Reply:
yes, it should not matter as we only check for non-nil error in the caller